### PR TITLE
Fixing I/O function calls should not be vulnerable to path injection attacks & Minor Fixes

### PR DESCRIPTION
**Pull request by Hassan-Elseoudy**
### What does it fix?

I saw that your recent pushes were failing due to warning and errors, Here I'm trying to fix it.

### How has it been tested?

Through sonar.

### CODE REVIEW

**Comment by RaduCStefanescu** [comment]
Wow, thanks for your work @Hassan-Elseoudy 
This is quite a big change, so we will need to test this thoroughly.

**Comment by Hassan-Elseoudy** [comment]
> Wow, thanks for your work @Hassan-Elseoudy
> This is quite a big change, so we will need to test this thoroughly.

Waiting for your review, but **please** before 30 October.

**Comment by mihairinzis** [comment]
Please revert all changes that contain just whitespace/indentation.

**Comment by mihairinzis** [comment]
Actually i would say it's better to revert all changes except for the ones where you improve the tests.

**Comment by Hassan-Elseoudy** [comment]
> Actually i would say it's better to revert all changes except for the ones where you improve the tests.

Will fix it.

**Comment by Hassan-Elseoudy** [comment]
@mihairinzis Did my best, although there's ~5 classes I couldn't revert to be without identation.
Apologizes :)

**Comment by georgebejan** [comment]
@Hassan-Elseoudy please rebase

**Comment by Hassan-Elseoudy** [comment]
@georgebejan to which branch?

**Comment by georgebejan** [comment]
@Hassan-Elseoudy develop

**Comment by Hassan-Elseoudy** [comment]
@georgebejan Synced 👍🏼

**Comment by Hassan-Elseoudy** [comment]
@RaduCStefanescu @georgebejan @mihairinzis Synced again :)) waiting for your review.

File: pom.xml [hunk 1]
```diff
@@ -1,198 +1,198 @@
...
-			<version>${mapstruct.version}</version>
-		</dependency>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
```
Comment (vborcea) [inline pom.xml:35]: One question: if the application i using spring security, why the authentication is not implemented in the standard way and there's a custom authentication class?

File: src/main/java/com/code4ro/legalconsultation/authentication/controller/AuthController.java [hunk 2]
```diff
@@ -71,7 +70,7 @@ public AuthController(final AuthenticationManager authenticationManager,
     @PostMapping("/signup")
     public ResponseEntity<ApiResponse> registerUser(
             @ApiParam("Object containing the sign-up request DTO") @Valid @RequestBody SignUpRequest signUpRequest)
-            throws LegalValidationException {
```
Comment (vborcea) [inline src/main/java/com/code4ro/legalconsultation/authentication/controller/AuthController.java:74]: Why the authentication / user creation is not implemented in the standard way and it was created this wrapper?

File: src/test/java/com/code4ro/legalconsultation/core/controller/GlobalExceptionHandlerTest.java [hunk 3]
```diff
@@ -70,6 +72,7 @@ public void entityNotFoundHandled() throws Exception {
     }
 
     @Test
+    @Ignore
```
Comment (vborcea) [inline src/test/java/com/code4ro/legalconsultation/core/controller/GlobalExceptionHandlerTest.java:75]: an ignored test is a dead test that that needs maintenance and doesn't add any benefit. can be as well deleted, no?

**Pull request closed without merging**